core/src/main/scala/org/apache/spark/deploy/master/ApplicationInfo.scala

```diff
@@ -96,6 +101,14 @@ private[spark] class ApplicationInfo(
 
   private[master] def coresLeft: Int = requestedCores - coresGranted
 
+  /**
+   * Return whether this application should launch at most one executor per worker.
+   *
+   * This is true if cores per executor is not defined, in which case the executor should
+   * grab all the available cores on the worker instead.
+   */
+  private[master] def oneExecutorPerWorker(): Boolean = desc.coresPerExecutor.isEmpty
+
```

Comment:
Is it better of being declared as a local variable in scheduleExecutorsOnWorkers? The decision of translating coresPerExecutor.isEmpty to oneExecutorPerWorker seems to belong to the scheduler, in case we decide to change default behavior in the future.